### DISCUSSION TITLE
how to change encryption key?

### DISCUSSION THREAD

**Discussion by sardine-antics** [zbm-dev/zfsbootmenu#460] [General]
Hi all,

New to zfsbootmenu and experimenting on a laptop.  It's already saved me a few times with the wonderful previous boot options.

I installed ubuntu with encryption.  The passphrase is kept in /etc/zfs/zroot.key.  Tried changing the file and played with zfs change-key, but ended up at the initramfs prompt.

Is there a way to change this key so everything boots smoothly?

thanks

**Comment by zdykstra** [comment]
With out knowing the exact steps you took and in what order, I can really only guess. However, it's probable that after changing the encryption key for the encryptionroot, the Ubuntu initramfs wasn't rebuilt to include the updated file. You'll need to recreate the initramfs for the kernel you want to boot so that it pulls in the corrected /etc/zfs/zroot.key . If you hit `MOD+W` on the main screen, your pool will be imported read-write so that you can chroot into your boot environment and make changes to files there.

**Reply by sardine-antics** [reply]
Many thanks!  Using the MOD+W to get the read-write chroot thing worked wonderfully.  Then ran update-initramfs and life was good.

Also while doing that I found I didn't have the bootfs set properly.  It's great to be able to play with the file system knowing it can be rolled back.  It encourages all kinds of sillyness.

Thanks to all you good folks for creating an excellent way to have the complete zfs ecosystem on linux.

much appreciated
